### PR TITLE
4.x CI tests fixes

### PR DESCRIPTION
**Pull request by avkos**
## Description
#5523

- [x] #5535
https://github.com/web3/web3.js/actions/runs/3223942245/jobs/5274588157
- [x] https://github.com/web3/web3.js/issues/5510 (@luu-alex)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-eth-contract/test/integration/contract_erc721.test.ts [hunk 1]
```diff
@@ -70,6 +71,15 @@ describe('contract', () => {
 				contractDeployed = await contract.deploy(deployOptions).send(sendOptions);
 			});
 
+			const getTokenId = async (res: Receipt) => {
+				const topic = (res.logs as EventLog[])[0]?.topics[0];
+				const logs = await contractDeployed.getPastEvents('Transfer');
+
+				return toBigInt(
+					(logs.find(l => (l as EventLog).topics.includes(topic)) as EventLog)
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-contract/test/integration/contract_erc721.test.ts]: It seems that this error: https://github.com/web3/web3.js/actions/runs/3269305208/jobs/5386153617 happens when the value is undefined. 

So, do you think it is a good solution to try to fetch the past event of 'Transfer' again if  `tokenId` was undefined?

**Review by avkos — COMMENTED** [review]
in this test, we deploy the contract in the section beforeEach. So it means every new test has a new contract instance. In this case `tokenId` will be `0n` always

**Review by avkos — COMMENTED** [review]
Also, I've fixed the timeout problem with the IPC provider in syncing event while deploying the contract

**Comment by avkos** [comment]
#5510 is not reproduced. I suggest to close it

File: packages/web3-eth-contract/test/integration/contract_erc721.test.ts [hunk 2]
```diff
@@ -136,8 +133,8 @@ describe('contract', () => {
 							),
 							pkAccount.privateKey,
 						);
-						const logs = await contractDeployed.getPastEvents('Transfer');
-						const tokenId = (logs[0] as EventLog)?.returnValues?.tokenId as string;
+						// eslint-disable-next-line
```
Comment (Spacesai1or) [inline packages/web3-eth-contract/test/integration/contract_erc721.test.ts]: The linter should be complaining about this, but you should provide the rule you're intending to disable

File: packages/web3-eth-contract/test/integration/setup.js [hunk 3]
```diff
@@ -19,6 +19,6 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 // eslint-disable-next-line @typescript-eslint/no-require-imports
 require('../config/setup');
 
-const jestTimeout = 15000;
+const jestTimeout = String(process.env.WEB3_SYSTEM_TEST_PROVIDER).includes('ipc') ? 25000 : 15000;
```
Comment (Spacesai1or) [inline packages/web3-eth-contract/test/integration/setup.js:22]: I would think creating a `JEST_TIMEOUT` (defaulting to `15000`) ENV would be a better approach than this as it's more explicit to the developer that the environment for IPC testing is different

**Review by Spacesai1or — CHANGES_REQUESTED** [review]
Everything else looks good, thank you for solving this issue!

File: packages/web3-eth-contract/test/integration/contract_erc721.test.ts (hunk 2 above)
Comment (avkos) [reply packages/web3-eth-contract/test/integration/contract_erc721.test.ts]: done

**Review by Spacesai1or — APPROVED** [review]
We can create an issue to track my [suggestion](https://github.com/web3/web3.js/pull/5544/files#r1000000384) if the team decides to move forward with it

**Review by jdevcs — APPROVED** [review]

File: packages/web3-eth-contract/test/integration/setup.js (hunk 3 above)
Comment (avkos) [reply packages/web3-eth-contract/test/integration/setup.js:22]: it is only for contract tests that's why I use this solution

**Pull request merged**